src/DynamoCore/Configuration/PreferenceSettings.cs

```diff
@@ -768,5 +751,81 @@ private static void MigrateStdLibTokenToBuiltInToken(PreferenceSettings settings
                 }
             }
         }
+
+        #region Trust Management API
+        /// <summary>
+        /// Add a path to the Dynamo's trusted locations
+        /// </summary>
+        /// <param name="path">The path to be added as a trusted location</param>
+        /// <returns></returns>
+        internal bool AddTrustedLocation(string path)
+        {
+            try
+            {
+                string location = PathHelper.ValidateDirectory(path);
+                var existing = TrustedLocations.FirstOrDefault(x => x.Equals(location));
+                if (string.IsNullOrEmpty(existing))
+                {
+                    trustedLocations.Add(location);
+                    return true;
+                }
+            }
+            catch { }
```

Comment:
We may want to use Dynamo logger to log the exception here

Reply:
yeah agreed, I think the easiest way to do this is the implement ILogSource and then hook up the events in the DynamoModel right after PreferenceSetting is constructed.